data/settings/src/main/java/com/google/jetpackcamera/settings/LocalSettingsRepository.kt

```diff
@@ -61,7 +69,18 @@ class LocalSettingsRepository @Inject constructor(
         }
     }
 
-    override suspend fun getSettings(): Settings {
-        return settings.first()
+    override suspend fun updateFlashModeStatus(flashModeStatus: FlashModeStatus) {
+        val newStatus = when (flashModeStatus) {
+            FlashModeStatus.AUTO -> FlashModeProto.FLASH_MODE_AUTO
+                FlashModeStatus.ON -> FlashModeProto.FLASH_MODE_ON
+                FlashModeStatus.OFF -> FlashModeProto.FLASH_MODE_OFF
+        }
+        jcaSettings.updateData {
+            it.copy { this.flashModeStatus = newStatus }
+        }
+    }
+
+    override suspend fun getSettings(): CameraAppSettings {
```

Comment:
nit: you can use expression function here since it's a simple expression
https://kotlinlang.org/docs/functions.html#single-expression-functions

`override suspend fun getSettings(): CameraAppSettings = cameraAppSettings.first()`